File: java/quantile-estimator/src/main/java/org/mydotey/quantileestimator/value/IntCalculator.java
package org.mydotey.quantileestimator.value;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
public class IntCalculator implements Calculator<Integer> {

    public static final IntCalculator DEFAULT = new IntCalculator();

    @Override
    public Integer add(Integer t, Integer t2) {
        return t + t2;
    }

    @Override
    public Integer subtract(Integer t, Integer t2) {
        return t - t2;
    }

    @Override
    public Integer multiply(Integer t, double t2) {
        return (int) (t * t2);
    }

    @Override
    public Integer divide(Integer t, double t2) {
        return (int) (t / t2);
    }

}

File: dotnet/src/MyDotey.QuantileEstimator/Value/IntCalculator.cs
using System;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
namespace MyDotey.Quantile.Value
{
    public class IntCalculator : ICalculator<int>
    {
        public static readonly IntCalculator DEFAULT = new IntCalculator();

        public virtual int Add(int t, int t2)
        {
            return t + t2;
        }

        public virtual int Subtract(int t, int t2)
        {
            return t - t2;
        }

        public virtual int Multiply(int t, double t2)
        {
            return (int)(t * t2);
        }

        public virtual int Divide(int t, double t2)
        {
            return (int)(t / t2);
        }
    }
}